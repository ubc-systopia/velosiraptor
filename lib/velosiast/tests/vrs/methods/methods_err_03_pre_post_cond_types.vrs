// wrong type from parameter
fn foo(foo: int, bar: int) -> int
    requires foo
{
    1
}

// wrong type doesn't exist
fn foo(foo: int, bar: int) -> int
    requires foobar
{
    1
}

// wrong type
fn foo(foo: int, bar: int) -> int
    requires foo & 0x1
{
    1
}

// ensures: they are currently ignored