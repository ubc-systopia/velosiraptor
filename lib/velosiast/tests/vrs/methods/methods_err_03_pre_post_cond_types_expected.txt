error: Requires clause has incompatible type
     --> tests/vrs/methods/methods_err_03_pre_post_cond_types.vrs:3:14
      |
    3 |      requires foo
      |               ^^^ Expected boolean, found int
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: no symbol named `foobar` found in the current scope
     --> tests/vrs/methods/methods_err_03_pre_post_cond_types.vrs:10:14
      |
   10 |      requires foobar
      |               ^^^^^^ not found in this scope
error: Requires clause has incompatible type
     --> tests/vrs/methods/methods_err_03_pre_post_cond_types.vrs:10:14
      |
   10 |      requires foobar
      |               ^^^^^^ Expected boolean, found int
error: could not compile due to 2 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: Requires clause has incompatible type
     --> tests/vrs/methods/methods_err_03_pre_post_cond_types.vrs:17:14
      |
   17 |      requires foo & 0x1
      |               ^^^^^^^^^ Expected boolean, found int
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

